lock.json

```diff
@@ -129,24 +116,21 @@
           "artifacts": [
             {
               "algorithm": "sha256",
-              "hash": "08706bdad8d0a3413266ab61bd6c34d0c28d6e1e7badf40a2cebe67644e2e1fb",
-              "url": "https://files.pythonhosted.org/packages/27/45/377f7e32a5c93d94cd56542349b34efab5ca3f9e2fd5a68c5e93169aa32d/Babel-2.15.0-py3-none-any.whl"
+              "hash": "368b5b98b37c06b7daf6696391c3240c938b37767d4584413e8438c5c435fa8b",
+              "url": "https://files.pythonhosted.org/packages/ed/20/bc79bc575ba2e2a7f70e8a1155618bb1301eaa5132a8271373a6903f73f8/babel-2.16.0-py3-none-any.whl"
             },
             {
               "algorithm": "sha256",
-              "hash": "8daf0e265d05768bc6c7a314cf1321e9a123afc328cc635c18622a2f30a04413",
-              "url": "https://files.pythonhosted.org/packages/15/d2/9671b93d623300f0aef82cde40e25357f11330bdde91743891b22a555bed/babel-2.15.0.tar.gz"
+              "hash": "d1f3554ca26605fe173f3de0c65f750f5a42f924499bf134de6423582298e316",
+              "url": "https://files.pythonhosted.org/packages/2a/74/f1bc80f23eeba13393b7222b11d95ca3af2c1e28edca18af487137eefed9/babel-2.16.0.tar.gz"
             }
           ],
           "project_name": "babel",
           "requires_dists": [
-            "freezegun~=1.0; extra == \"dev\"",
-            "pytest-cov; extra == \"dev\"",
-            "pytest>=6.0; extra == \"dev\"",
-            "pytz>=2015.7; python_version < \"3.9\""
+            "pytest>=6.0; extra == \"dev\""
```

Comment:
Just to clarify my understanding of the pex feature: 

`freezegun` and `pytest-cov` were filtered out because the `dev` extra of `babel` was not (transitively) requested via the input requirements? What I'm actually asking is, why is `pytest` is still here?

Reply:
The `pytest` dep is still there because its an input requirement:
https://github.com/a-scie/lift/blob/9edff13c1ce9fe7fa8a9565f55f7a338f0872713/lock.json#L1865-L1883

That's a bug though since the `dev` extra is never activated. In other words, the feature currently doesn't elide enough and you can be left with 1/2 an extra set. If you then subsetted via `pex --lock lock.json foo[dev] -o foo.pex` you'd only get part of the `dev` extra, which is certainly wrong. All or nothing seem correct, and since the `dev` extra was not requested via the input requirements, it should be nothing.

I'll follow up both in Pex and here after.

Reply:
Yeah, that's what I was grasping at. Even if pytest can be brought in by some other dep path, there is no need to mention it here.

Reply:
Ok, fixed up in #105.